File: frontend/src/types/feishu.ts
export interface FeishuConfig {
  app_id: string | null
  has_app_secret: boolean
  bot_open_id: string | null
  default_push_chat_id: string | null
  ws_connected: boolean
  enabled: boolean
}

export interface FeishuConfigUpdate {
  app_id?: string
  app_secret?: string
  bot_open_id?: string
  default_push_chat_id?: string
  enabled?: boolean
}

export interface FeishuPushRequest {
  chat_id: string
  message: string
  receive_id_type?: string
}

export interface FeishuCreateGroupRequest {
  name: string
  user_open_ids: string[]
  description?: string
}

File: backend/app/schemas/feishu.py
from typing import Optional
from pydantic import BaseModel


class FeishuConfigUpdate(BaseModel):
    app_id: Optional[str] = None
    app_secret: Optional[str] = None       # plaintext, encrypted on write
    bot_open_id: Optional[str] = None
    default_push_chat_id: Optional[str] = None
    enabled: Optional[bool] = None


class FeishuConfigRead(BaseModel):
    app_id: Optional[str] = None
    has_app_secret: bool = False
    bot_open_id: Optional[str] = None
    default_push_chat_id: Optional[str] = None
    ws_connected: bool = False             # runtime status: is WS thread alive?
    enabled: bool = False

    model_config = {"from_attributes": True}


class FeishuPushRequest(BaseModel):
    chat_id: str
    message: str
    receive_id_type: str = "chat_id"


class FeishuCreateGroupRequest(BaseModel):
    name: str
    user_open_ids: list[str]
    description: str = ""
